lib/iris/cube.py

```diff
@@ -790,6 +800,285 @@ def _is_single_item(testee):
     return isinstance(testee, str) or not isinstance(testee, Iterable)
 
 
+class CubeAttrsDict(MutableMapping):
+    """
+    A dict-like object for "Cube.attributes", which provides unified user access to
+    the combined cube 'local' and 'global' attributes, mimicking the behaviour of a
+    simple dict.
+
+    This supports all the regular methods of a 'dict'.
+    However, a few things such as the detailed print (repr) are different.
+
+    In addition, the 'locals' and 'globals' properties provide specific access to local
+    and global attributes, as regular
+    :class:`~iris.common.mixin.LimitedAttributeDict`\\s.
+
+    Notes
+    -----
+    For type testing, "issubclass(CubeAttrsDict, Mapping)" is True, but
+    "issubclass(CubeAttrsDict, dict)" is False.
+
+    Properties 'locals' and 'globals' are the two sets of cube attributes.  These are
+    both :class:`~iris.common.mixin.LimitedAttributeDict`.  The CubeAttrsDict object
+    contains *no* additional state of its own, but simply acts as a view on these two.
+
+    When reading (__getitem__, pop, popitem, keys, values etc), it contains all the
+    keys + values of both 'locals' and 'globals'.  When a key occurs in *both* 'locals'
+    and 'globals', the result is the local value.
+    See :meth:``~iris.cube.CubeAttrsDict__getitem__``.
+
+    When writing (__setitem__, setdefault, update, etc) to a key already present, the
+    existing entry in either 'locals' or 'globals' is updated.  If both are present,
+    'locals' is updated.  See :meth:``~iris.cube.CubeAttrsDict__setitem__``.
```

Comment:
```suggestion
    See :meth:`~iris.cube.CubeAttrsDict.__getitem__`.

    When writing (__setitem__, setdefault, update, etc) to a key already present, the
    existing entry in either 'locals' or 'globals' is updated.  If both are present,
    'locals' is updated.  See :meth:`~iris.cube.CubeAttrsDict.__setitem__`.
```

This may be pointless though as I don't think these methods get rendered.